### ISSUE TITLE
rake default should not depend on host system config

### ISSUE THREAD

**Issue by tejasbubane** [rubocop-hq/rubocop#7032]
I had some invalid config lying around in my old system in `~/.rubocop.yml`. Running `bundle exec rake default` in [rubocop project](https://github.com/rubocop-hq/rubocop) (before contributing a fix) fails.

--------

## Expected behavior

`rake default` should use its own config rather than depending on host system config. Disabled cops in host system can lead to inconsistent checks in local and CI.

## Actual behavior

Rake task fails complaining about invalid config file in home directory.

```
$ bundle exec rake default
....<log from other checks skipped>
Running RuboCop...
Error: The `Layout/IndentHash` cop has been renamed to `Layout/IndentFirstHashElement`.
(obsolete configuration found in /Users/tejas.bubane/.rubocop.yml, please update it)
RuboCop failed!
```

## Steps to reproduce the problem

Create `~/.rubocop.yml` with following content:

```yml
Layout/IndentHash:
  Enabled: false
```

Run `bundle exec rake default` in [rubocop project](https://github.com/rubocop-hq/rubocop) directory.

## RuboCop version

```
$ [bundle exec] rubocop -V
0.68.1 (using Parser 2.6.3.0, running on ruby 2.6.3 x86_64-darwin18)
```

**Comment by behraaang** [comment]
Hello there 
I am trying to get this resolved, but in the Rakefile there is nothing mentioned about using the default.yml file config if the argument is `default`, is that the required behavior? and can I do this if I got that right ?

**Comment by vfonic** [comment]
I've had created a duplicated issue: https://github.com/rubocop-hq/rubocop/issues/7132
So I'm closing the other issue to keep this in one place.

Just to add, this doesn't only happen on rubocop project, but on any project.

Here's the issue's content:

<details>

My `~/.rubocop.yml` requires `rubocop-rails`. However, when I tried to run `bundle exec rake default` (essentially rubocop) in rubocop git repo's root, I received the following error:

```sh
$ bundle exec rake default
Running RuboCop...
cannot load such file -- rubocop-rails
/Users/viktor/Developer/gems/rubocop/lib/rubocop/config_loader_resolver.rb:15:in `require'
/Users/viktor/Developer/gems/rubocop/lib/rubocop/config_loader_resolver.rb:15:in `block in resolve_requires'
/Users/viktor/Developer/gems/rubocop/lib/rubocop/config_loader_resolver.rb:11:in `each'
/Users/viktor/Developer/gems/rubocop/lib/rubocop/config_loader_resolver.rb:11:in `resolve_requires'
/Users/viktor/Developer/gems/rubocop/lib/rubocop/config_loader.rb:45:in `load_file'
/Users/viktor/Developer/gems/rubocop/lib/rubocop/config_loader.rb:107:in `add_excludes_from_files'
/Users/viktor/Developer/gems/rubocop/lib/rubocop/config_loader.rb:92:in `configuration_from_file'
/Users/viktor/Developer/gems/rubocop/lib/rubocop/config_store.rb:44:in `for'
/Users/viktor/Developer/gems/rubocop/lib/rubocop/cli.rb:180:in `validate_options_vs_config'
/Users/viktor/Developer/gems/rubocop/lib/rubocop/cli.rb:48:in `run'
/Users/viktor/Developer/gems/rubocop/lib/rubocop/rake_task.rb:44:in `run_cli'
/Users/viktor/Developer/gems/rubocop/lib/rubocop/rake_task.rb:28:in `block (2 levels) in initialize'
/Users/viktor/.rvm/rubies/ruby-2.6.3/lib/ruby/gems/2.6.0/gems/rake-12.3.2/lib/rake/file_utils_ext.rb:59:in `verbose'
/Users/viktor/Developer/gems/rubocop/lib/rubocop/rake_task.rb:26:in `block in initialize'
/Users/viktor/.rvm/rubies/ruby-2.6.3/lib/ruby/gems/2.6.0/gems/rake-12.3.2/lib/rake/task.rb:273:in `block in execute'
/Users/viktor/.rvm/rubies/ruby-2.6.3/lib/ruby/gems/2.6.0/gems/rake-12.3.2/lib/rake/task.rb:273:in `each'
/Users/viktor/.rvm/rubies/ruby-2.6.3/lib/ruby/gems/2.6.0/gems/rake-12.3.2/lib/rake/task.rb:273:in `execute'
/Users/viktor/.rvm/rubies/ruby-2.6.3/lib/ruby/gems/2.6.0/gems/rake-12.3.2/lib/rake/task.rb:214:in `block in invoke_with_call_chain'
/Users/viktor/.rvm/rubies/ruby-2.6.3/lib/ruby/2.6.0/monitor.rb:230:in `mon_synchronize'
/Users/viktor/.rvm/rubies/ruby-2.6.3/lib/ruby/gems/2.6.0/gems/rake-12.3.2/lib/rake/task.rb:194:in `invoke_with_call_chain'
/Users/viktor/.rvm/rubies/ruby-2.6.3/lib/ruby/gems/2.6.0/gems/rake-12.3.2/lib/rake/task.rb:238:in `block in invoke_prerequisites'
/Users/viktor/.rvm/rubies/ruby-2.6.3/lib/ruby/gems/2.6.0/gems/rake-12.3.2/lib/rake/task.rb:236:in `each'
/Users/viktor/.rvm/rubies/ruby-2.6.3/lib/ruby/gems/2.6.0/gems/rake-12.3.2/lib/rake/task.rb:236:in `invoke_prerequisites'
/Users/viktor/.rvm/rubies/ruby-2.6.3/lib/ruby/gems/2.6.0/gems/rake-12.3.2/lib/rake/task.rb:213:in `block in invoke_with_call_chain'
/Users/viktor/.rvm/rubies/ruby-2.6.3/lib/ruby/2.6.0/monitor.rb:230:in `mon_synchronize'
/Users/viktor/.rvm/rubies/ruby-2.6.3/lib/ruby/gems/2.6.0/gems/rake-12.3.2/lib/rake/task.rb:194:in `invoke_with_call_chain'
/Users/viktor/.rvm/rubies/ruby-2.6.3/lib/ruby/gems/2.6.0/gems/rake-12.3.2/lib/rake/task.rb:183:in `invoke'
/Users/viktor/.rvm/rubies/ruby-2.6.3/lib/ruby/gems/2.6.0/gems/rake-12.3.2/lib/rake/application.rb:160:in `invoke_task'
/Users/viktor/.rvm/rubies/ruby-2.6.3/lib/ruby/gems/2.6.0/gems/rake-12.3.2/lib/rake/application.rb:116:in `block (2 levels) in top_level'
/Users/viktor/.rvm/rubies/ruby-2.6.3/lib/ruby/gems/2.6.0/gems/rake-12.3.2/lib/rake/application.rb:116:in `each'
/Users/viktor/.rvm/rubies/ruby-2.6.3/lib/ruby/gems/2.6.0/gems/rake-12.3.2/lib/rake/application.rb:116:in `block in top_level'
/Users/viktor/.rvm/rubies/ruby-2.6.3/lib/ruby/gems/2.6.0/gems/rake-12.3.2/lib/rake/application.rb:125:in `run_with_threads'
/Users/viktor/.rvm/rubies/ruby-2.6.3/lib/ruby/gems/2.6.0/gems/rake-12.3.2/lib/rake/application.rb:110:in `top_level'
/Users/viktor/.rvm/rubies/ruby-2.6.3/lib/ruby/gems/2.6.0/gems/rake-12.3.2/lib/rake/application.rb:83:in `block in run'
/Users/viktor/.rvm/rubies/ruby-2.6.3/lib/ruby/gems/2.6.0/gems/rake-12.3.2/lib/rake/application.rb:186:in `standard_exception_handling'
/Users/viktor/.rvm/rubies/ruby-2.6.3/lib/ruby/gems/2.6.0/gems/rake-12.3.2/lib/rake/application.rb:80:in `run'
/Users/viktor/.rvm/rubies/ruby-2.6.3/lib/ruby/gems/2.6.0/gems/rake-12.3.2/exe/rake:27:in `<top (required)>'
/Users/viktor/.rvm/rubies/ruby-2.6.3/bin/rake:23:in `load'
/Users/viktor/.rvm/rubies/ruby-2.6.3/bin/rake:23:in `<main>'
/Users/viktor/.rvm/gems/ruby-2.6.3/bin/ruby_executable_hooks:24:in `eval'
/Users/viktor/.rvm/gems/ruby-2.6.3/bin/ruby_executable_hooks:24:in `<main>'
RuboCop failed!
```

## Steps to reproduce the problem

```sh
mv ~/.rubocop.yml ~/.rubocop.yml_bak # backup your rubocop config if any
echo "require: 'rubocop-rails'" > ~/.rubocop.yml
rubocop # or bundle exec rake default, depending where you are
```

Since rubocop repo has its own `.rubocop.yml`, the `~/.rubocop.yml` in my home directory should be completely ignored.

This makes it really hard to work on any project that doesn't have `rubocop-rails` as a dependency, even if it has its own `.rubocop.yml` config file.

## RuboCop version

```
$ [bundle exec] rubocop -V
0.71.0 (using Parser 2.6.3.0, running on ruby 2.6.3 x86_64-darwin18)
```

</details>

**Comment by buehmann** [comment]
I can reproduce this as and tried to reduce the problem via `bundle exec rake internal_investigation` down to
```
➜  rubocop (master) bundle exec rubocop --debug
...
AllCops/Exclude configuration from /Users/buehmann/.rubocop.yml
Error: The `Layout/IndentHash` cop has been renamed to `Layout/IndentFirstHashElement`.
(obsolete configuration found in /Users/buehmann/.rubocop.yml, please update it)
...
```
The `AllCops/Exclude` line leads to [this spec](https://github.com/rubocop-hq/rubocop/blob/f04ad39050717a95cd8d2b7075f8c96b50d94342/spec/rubocop/config_loader_spec.rb#L136-L156) which was introduced in https://github.com/rubocop-hq/rubocop/pull/295
> Get config parameter AllCops/Excludes from highest config file in path

Sure enough, my rubocop checkout is in a directory below my home directory:
```
➜  rubocop (master) pwd
/Users/buehmann/dev/rubocop
```
So that's what happens … Not sure what the right solution is.

**Comment by buehmann** [comment]
Adding `--ignore-parent-exclusion` to the rubocop options in `Rakefile` would be one option. See https://github.com/rubocop-hq/rubocop/issues/4328

**Comment by tejasbubane** [comment]
@buehmann I already tried it, does not work. It only ignores `Exclude` from parent. My parent config has an invalid cop name. Not sure but I think we might need another flag which disables parent inheritance entirely.

**Comment by buehmann** [comment]
You're right, my mistake. `--ignore-parent-exclusion` still loads `~/.rubocop.yml`.

**Comment by jonas054** [comment]
Does it? Not for me. When I add `--ignore-parent-exclusion` the problem goes away.
```
 /tmp/7032$ (find . -type f; echo ~/.rubocop.yml) | xargs head
==> ./.rubocop.yml <==
AllCops:
  DisabledByDefault: true

==> ./example.rb <==
puts 1

==> /home/jonas/.rubocop.yml <==
Invalid
 /tmp/7032$ rubocop -d
For /tmp/7032: configuration from /tmp/7032/.rubocop.yml
Default configuration from /usr/local/rvm/gems/ruby-2.6.3/gems/rubocop-0.82.0/config/default.yml
AllCops/Exclude configuration from /home/jonas/.rubocop.yml
Malformed configuration in /home/jonas/.rubocop.yml
/usr/local/rvm/gems/ruby-2.6.3/gems/rubocop-0.82.0/lib/rubocop/config_loader.rb:227:in `load_yaml_configuration'
/usr/local/rvm/gems/ruby-2.6.3/gems/rubocop-0.82.0/lib/rubocop/config_loader.rb:42:in `load_file'
/usr/local/rvm/gems/ruby-2.6.3/gems/rubocop-0.82.0/lib/rubocop/config_loader.rb:117:in `add_excludes_from_files'
/usr/local/rvm/gems/ruby-2.6.3/gems/rubocop-0.82.0/lib/rubocop/config_loader.rb:93:in `configuration_from_file'
/usr/local/rvm/gems/ruby-2.6.3/gems/rubocop-0.82.0/lib/rubocop/config_store.rb:44:in `for'
/usr/local/rvm/gems/ruby-2.6.3/gems/rubocop-0.82.0/lib/rubocop/cli.rb:126:in `apply_default_formatter'
/usr/local/rvm/gems/ruby-2.6.3/gems/rubocop-0.82.0/lib/rubocop/cli.rb:40:in `run'
/usr/local/rvm/gems/ruby-2.6.3/gems/rubocop-0.82.0/exe/rubocop:13:in `block in <top (required)>'
/usr/local/rvm/rubies/ruby-2.6.3/lib/ruby/2.6.0/benchmark.rb:308:in `realtime'
/usr/local/rvm/gems/ruby-2.6.3/gems/rubocop-0.82.0/exe/rubocop:12:in `<top (required)>'
/usr/local/rvm/gems/ruby-2.6.3/bin/rubocop:23:in `load'
/usr/local/rvm/gems/ruby-2.6.3/bin/rubocop:23:in `<main>'
/usr/local/rvm/gems/ruby-2.6.3/bin/ruby_executable_hooks:24:in `eval'
/usr/local/rvm/gems/ruby-2.6.3/bin/ruby_executable_hooks:24:in `<main>'
Finished in 0.0699394969997229 seconds
 /tmp/7032$ rubocop -d --ignore-parent-exclusion
For /tmp/7032: configuration from /tmp/7032/.rubocop.yml
Default configuration from /usr/local/rvm/gems/ruby-2.6.3/gems/rubocop-0.82.0/config/default.yml
Ignoring AllCops/Exclude from parent foldersInspecting 1 file
Scanning /tmp/7032/example.rb
.

1 file inspected, no offenses detected
Finished in 0.11776560500220512 seconds
```

**Comment by tejasbubane** [comment]
@jonas054 I am unable to reproduce this issue with rubocop `0.82.0`. Not sure when & how this got fixed though. Closing.
